expapi/v2/types/metrics_types.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"time"
+
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/util"
+)
+
+type MetricPoint struct {
+	Timestamp time.Time `json:"timestamp"`
+	Value     uint64    `json:"value"`
+}
+
+type MetricResult struct {
+	Metrics         []MetricPoint `json:"metrics"`
+	LatestTimestamp time.Time     `json:"latestTimestamp"`
+}
+
+type MetricResultList struct {
+	Items []MetricResult `json:"items"`
+}
+
+type DerivedNodeMetrics struct {
+	v1.TypeMeta   `json:",inline"`
+	v1.ObjectMeta `json:"metadata,omitempty"`
+
+	NodeMetrics      MetricsWindows
+	SystemContainers []struct {
+		ContainerName string         `json:"containername"`
+		Windows       MetricsWindows `json:"windows"`
+	}
+}
+
+type DerivedNodeMetricsList struct {
+	v1.TypeMeta `json:",inline"`
```

Comment:
Are the `meta` objects required from heapster perspective? Since it provides read-only APIs I don't see a reason. I could be wrong.